src/games/strategy/triplea/ui/BattleDisplay.java

```diff
@@ -303,73 +306,43 @@ public void changedUnitsNotification(final PlayerID player, final Collection<Uni
 
   protected void waitForConfirmation(final String message) {
     if (SwingUtilities.isEventDispatchThread()) {
-      throw new IllegalStateException("This cant be in dispatch thread");
+      throw new IllegalStateException("This can not be in dispatch thread");
     }
 
-    if (getConfirmDefensiveRolls()) {
-      waitForSpaceBarClick(message);
-    } else {
-      // Sleep for a short period, allows users to see the dice rolls before the game
-      // moves on.
-      // TODO: This value should find its way into a setting. For now it is hardcoded since
-      // just having another confusing value to set would not help very much. Needs to be some
-      // context/explanation around it for a user to really want to set it.
-      try {
-        Thread.sleep(1500);
-      } catch (InterruptedException e) {
-      }
-    }
-  }
-
-  private void waitForSpaceBarClick(final String message) {
     final CountDownLatch continueLatch = new CountDownLatch(1);
-    // set the action in the swing thread.
-    SwingUtilities.invokeLater(new Runnable() {
-      @Override
-      public void run() {
-        m_actionButton.setAction(new AbstractAction(message) {
-          private static final long serialVersionUID = 4489826259192394858L;
+    SwingUtilities.invokeLater(() -> m_actionButton.setAction(SwingAction.of(message, e -> continueLatch.countDown())));
```

Comment:
Would it be better to break this line up some? Its kind of hard to follow all the nesting in one line.